### PR TITLE
User controller

### PR DESCRIPTION
**Pull request by LinKassem**

### CODE REVIEW

File: app/Http/Controllers/UsersController.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    public function index()
+    {
+        $users = User::all();
+        return view('users.index', compact('users'));
+    }
+
+    public function show($id)
+    {
+        $user = User::findOrfail($id);
```
Comment (mansoursaid) [inline app/Http/Controllers/UsersController.php:19]: findorfail exception could be caught i think

File: app/Http/Controllers/UsersController.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    public function create()
+    {
+        return view('users.create');
+    }
+
+    public function store(Requests\CreateUserRequest $request)
+    {
+        // validations
```
Comment (mansoursaid) [inline app/Http/Controllers/UsersController.php:30]: dont forget these

File: app/Http/routes.php [hunk 3]
```diff
@@ -34,6 +33,15 @@
...
 
+Route::get('users','UsersController@index');
+Route::get('users/create','UsersController@create');
+Route::post('users','UsersController@store');
+Route::get('users/{id}','UsersController@show');
+Route::delete('users/{id}','UsersController@destroy');
+Route::get('users/{id}/edit','UsersController@edit');
+Route::PATCH('users/{id}','UsersController@update');
+Route::DELETE('users','UsersController@delete');
+
```
Comment (mansoursaid) [inline app/Http/routes.php:44]: use Route::resource
a 1 liner and generates all those for you
you can check the generated routes with php artisan route:list and adapt to the ones generated

File: app/User.php [hunk 4]
```diff
@@ -12,7 +12,7 @@ class User extends Authenticatable
      * @var array
      */
     protected $fillable = [
-        'name', 'email', 'password',
+        'name', 'email', 'password', 'type',
```
Comment (mansoursaid) [inline app/User.php:15]: I dont think the type should be a fillable in the user, for security reasons.
in my opinion, a type can only be set by an admin through his panel and all users are normal at first when they are registered to the system.
The first admin of the system should be supplied with the system when it is delivered but not created

**Comment by mansoursaid** [comment]
@LinKassem check my comments here

File: app/Http/Controllers/UsersController.php (hunk 1 above)
Comment (LinKassem) [inline app/Http/Controllers/UsersController.php:19]: @mansoursaid001  So it should be like this right?

 public function show($id)
    {
        // Will return a ModelNotFoundException if no user with that id
        try
        {
            $user = User::findOrFail($id);
        }
        // catch(Exception $e) catch any exception
        catch(ModelNotFoundException $e)
        {
            // lists all available methods for exception object
            // Note that: dd($value) Dumps the $variable and end execution of the script.
            dd(get_class_methods($e))
            dd($e)
        }

```
    return view('users.show', compact('user'));
}
```

Comment (LinKassem) [inline app/Http/Controllers/UsersController.php:19]: @mansoursaid001  Also do you know any other alternative for the try&catch block? I searching but I thought I should also ask you.

File: app/Http/Controllers/UsersController.php (hunk 2 above)
Comment (LinKassem) [inline app/Http/Controllers/UsersController.php:30]: @mansoursaid001  Can you be a little bit more clear please. Thank you

File: app/Http/routes.php (hunk 3 above)
Comment (LinKassem) [inline app/Http/routes.php:44]: (Y)

File: app/User.php (hunk 4 above)
Comment (LinKassem) [inline app/User.php:15]: Okay (Y)

**Pull request merged**